xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/url/resources/DocumentRestURLGenerator.java

```diff
@@ -44,6 +46,13 @@ public URL getURL(DocumentReference reference) throws XWikiRestException
         // The idea is to use the UriBuilder of jax-rs to generate URLs that match the resources paths.
         // So it is consistent.
         try {
+            Locale locale = reference.getLocale();
+            // Get the URL of the translated document if possible
+            if (locale != null && !locale.toString().isEmpty()) {
+                return Utils.createURI(getBaseURI(), PageTranslationResource.class,
+                    reference.getWikiReference().getName(), getSpaceList(reference.getLastSpaceReference()),
+                    reference.getName(), reference.getLocale()).toURL();
+            }
```

Comment:
Small detail: could you introduce some variable so that the methods only has a single "return" statement? That makes it simpler to read and less error prone. Thx